include/boost/url/detail/replacement_id_rule.hpp

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_DETAIL_REPLACEMENT_ID_RULE_HPP
+#define BOOST_URL_DETAIL_REPLACEMENT_ID_RULE_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/error.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/grammar/alnum_chars.hpp>
+#include <boost/url/grammar/alpha_chars.hpp>
+#include <boost/url/grammar/delim_rule.hpp>
+#include <boost/url/grammar/lut_chars.hpp>
+#include <boost/url/grammar/optional_rule.hpp>
+#include <boost/url/grammar/token_rule.hpp>
+#include <boost/url/grammar/tuple_rule.hpp>
+#include <boost/url/grammar/unsigned_rule.hpp>
+#include <boost/url/grammar/variant_rule.hpp>
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+// replacement_field ::=  "{" [arg_id] [":" format_spec "}"
+struct replacement_id_rule_t
+{
+    using value_type = string_view;
+
+    BOOST_URL_DECL
```

Comment:
is this function called from user-visible code? If not, it doesn't need the dll export.

Reply:
We had an error in win32 because it's called from code header files I think. I don't remember which. I'll check.